library/alloc/src/rc.rs

```diff
@@ -1679,15 +1679,8 @@ impl<T: ?Sized> From<Box<T>> for Rc<T> {
 #[stable(feature = "shared_from_slice", since = "1.21.0")]
 impl<T> From<Vec<T>> for Rc<[T]> {
     #[inline]
-    fn from(mut v: Vec<T>) -> Rc<[T]> {
-        unsafe {
-            let rc = Rc::copy_from_slice(&v);
-
-            // Allow the Vec to free its memory, but not destroy its contents
-            v.set_len(0);
-
-            rc
-        }
+    fn from(v: Vec<T>) -> Rc<[T]> {
+        Rc::from_box(v.into_boxed_slice())
```

Comment:
`Rc::from_box` copies itself too. In addition if the length and capacity of the vector are not identical `Vec::into_boxed_slice` will do an additional copy, so this can only be either as fast or slower than the previous implementation depending on whether the length and capacity are identical. It has to copy one way or another to fit the reference counts before the data.

Reply:
> Rc::from_box copies itself too. 

And I missed this completely. Shame on me. Thanks!